src/style-spec/feature_filter/index.js

```diff
@@ -21,6 +21,8 @@ function isExpressionFilter(filter: any) {
         return filter.length >= 2 && filter[1] !== '$id' && filter[1] !== '$type';
 
     case 'in':
+        return filter.length === 3 && Array.isArray(filter[2]);
```

Comment:
I can imagine us adding a collator argument to `"in"`. We introduced [a bug](https://github.com/mapbox/mapbox-gl-js/issues/6920) when we we did that for string comparison operators. Let's future-proof this line by checking `filter.length >= 3`.